src/lib/transaction_snark/test/zkapps_examples/big_circuit/big_circuit.ml

```diff
@@ -0,0 +1,162 @@
+open Transaction_snark_tests.Util
+open Core_kernel
+open Mina_base
+open Signature_lib
+module Impl = Pickles.Impls.Step
+module Inner_curve = Snark_params.Tick.Inner_curve
+module Nat = Pickles_types.Nat
+module Local_state = Mina_state.Local_state
+module Zkapp_command_segment = Transaction_snark.Zkapp_command_segment
+
+(* Hacky way to detect exceptions being raised. *)
+module MustRaise = struct
+  exception WontCatch
+
+  let check (fn : unit -> unit) =
+    let wrapper _ = fn () ; raise WontCatch in
+    try wrapper () with e when Poly.(e <> WontCatch) -> ()
```

Comment:
If this feels too hacky for you, you can always
```ocaml
if Result.try_with fn |> Result.is_ok then raise VerySadVeryBad else ()
```

Reply:
oh that's much nicer! Thanks